can/io/asc.py

```diff
@@ -360,6 +361,13 @@ def __init__(
         :param channel: a default channel to use when the message does not
                         have a channel set
         """
+        if "append" in kwargs:
+            if kwargs["append"]:
+                raise TypeError(
```

Comment:
I'm not sure if this is the best kind of error to use. Maybe a `ValueError`?

Reply:
Yes, I'd say ValueError, too